src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -243,6 +249,17 @@ internal static ReadOnlySpan<byte> TrimEndingNullsAndSpaces(ReadOnlySpan<byte> b
             return buffer.Slice(0, trimmedLength);
         }
 
+        private static ReadOnlySpan<byte> TrimLeadingNullsAndSpaces(ReadOnlySpan<byte> buffer)
+        {
+            int newStart = 0;
+            while (newStart < buffer.Length && buffer[newStart] is 0 or 32)
+            {
+                newStart++;
+            }
+
+            return buffer.Slice(newStart);
+        }
```

Comment:
nit: you can use the new `IndexOfAnyExcept` here

```suggestion
        => buffer.Slice(Math.Max(0, buffer.IndexOfAnyExcept((byte)0, (byte)' '));
```

Reply:
Since it's a nit, I can address this in a separate PR.